### PR TITLE
Update pybind11 version to v2.8.1 release

### PR DESCRIPTION
**Pull request by eleftherioszisis**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	path = 3rdparty/HighFive
 	url = https://github.com/BlueBrain/HighFive.git
 [submodule "3rdparty/pybind11"]
-	path = 3rdparty/pybind11
+	path = binds/python/pybind11
```
Comment (mgeplf) [inline .gitmodules:5]: why the move here?

Comment (eleftherioszisis) [reply .gitmodules:5]: The way it is arranged currently, an if clause is needed in the `3rdparty/` to optionally add the `pybind11` subdirectory.

By moving it to the python dir, all the Python-related stuff is in the same place and only the root level inclusion of the subdirectory is needed in the case of building the wheels. It also reflects how libsonata is organized wrt the python bindings.

Comment (mgeplf) [reply .gitmodules:5]: I see, I think the `MANIFEST.in` probably needs to be updated then.

When making a change like this, it's worth checking the the sdist builds fine.

Comment (eleftherioszisis) [reply .gitmodules:5]: Good catch. I updated the `MANIFEST.in` and tested the sdist build.

File: MANIFEST.in [hunk 2]
```diff
@@ -17,5 +17,8 @@ include 3rdparty/pybind11/CMakeLists.txt
 include 3rdparty/lexertl14/CMakeLists.txt
 recursive-include 3rdparty/lexertl14/include/lexertl/ *
 
-recursive-include 3rdparty/pybind11/tools/ *
-include CMakeLists.txt
```
Comment (mgeplf) [inline MANIFEST.in:21]: confused how it works w/o the `CMakeLists.txt` file, but sdist is adding it 'magically'; let's make it explicit.

line 14/15 need to go

File: MANIFEST.in [hunk 3]
```diff
@@ -21,3 +21,12 @@ recursive-include binds/python/pybind11/include *
 recursive-include binds/python/pybind11/tools *.cmake
 include binds/python/pybind11/CMakeLists.txt
 include binds/python/pybind11/LICENSE
+
+# setuptools_scm forces all SCM files to be packaged into sdist
```
Comment (mgeplf) [inline MANIFEST.in:25]: Gorgeous!  ~1/10th the size:
```
-rw-r--r--  1  4535022 Dec 15 08:54 MorphIO-3.3.3.dev15+g16a7222.d20211215.tar.gz
-rw-r--r--  1   478010 Dec 15 08:55 MorphIO-3.3.3.dev17+gd64ba5a.d20211215.tar.gz
```

**Review by mgeplf — APPROVED** [review]

**Pull request merged**